### PR TITLE
OPSEXP-1416: controled usage of become plugin in ansible sync role

### PR DESCRIPTION
**Pull request by alxgomz**
OPSEXP-1416: controled usage of become plugin in ansible sync role

### CODE REVIEW

File: roles/sync/tasks/main.yml [hunk 1]
```diff
@@ -4,99 +4,70 @@
...
+    dest: "{{ download_location }}/sync-dist-6.x-{{ sync.version }}.zip"
+    checksum: sha1:{{ lookup('url', downloads.sync_zip_sha1_checksum_url, username=nexus_user, password=nexus_password) }} # pragma: allowlist secret
+    url_username: "{{ nexus_user }}"
+    url_password: "{{ nexus_password }}"
+
+- name: Extract Sync service to {{ sync_home }}
+  unarchive:
+    src: "{{ download_location }}/sync-dist-6.x-{{ sync.version }}.zip"
+    dest: "{{ download_location }}"
+    remote_src: true
```
Comment (gionn) [inline roles/sync/tasks/main.yml:19]:
```suggestion
```

File: roles/sync/tasks/main.yml [hunk 2]
```diff
@@ -4,99 +4,70 @@
...
-        dest: "{{ download_location }}/sync-dist-6.x-{{ sync.version }}.zip"
-        checksum: sha1:{{ lookup('url', downloads.sync_zip_sha1_checksum_url, username=nexus_user, password=nexus_password) }} # pragma: allowlist secret
+    - name: Ensure a list of packages installed
+      package:
+        name: "{{ utils_sync }}"
+        state: present
+
+    - name: Copy files to {{ sync_home }}
+      copy:
+        remote_src: true
```
Comment (gionn) [inline roles/sync/tasks/main.yml:32]:
```suggestion
```

File: roles/sync/tasks/main.yml [hunk 3]
```diff
@@ -108,58 +79,38 @@
...
-    - wait-for-repo
-    - restart-sync
+    - name: Add sync service startup script
+      template:
+        src: syncservice.sh
+        dest: "{{ binaries_folder }}/syncservice.sh"
+        owner: "{{ username }}"
+        group: "{{ group_name }}"
+        mode: 'u=rwx,g=rwx'
+        force: no
```
Comment (gionn) [inline roles/sync/tasks/main.yml:100]:
```suggestion
```

File: roles/sync/tasks/main.yml [hunk 4]
```diff
@@ -108,58 +79,38 @@
...
-  - enable-sync
-  - restart-sync
+    - name: Add alfresco-sync.service service
+      template:
+        src: alfresco-sync.service
+        dest: "/etc/systemd/system/alfresco-sync.service"
+        owner: "{{ username }}"
+        group: "{{ group_name }}"
+        mode: 'u=rwx,g=rwx'
+        force: yes
```
Comment (gionn) [inline roles/sync/tasks/main.yml:112]:
```suggestion
```

File: roles/sync/tasks/main.yml (hunk 1 above)
Comment (alxgomz) [reply roles/sync/tasks/main.yml:19]: I'm not sure I get your point.
To me this is needed: https://docs.ansible.com/ansible/latest/collections/ansible/builtin/unarchive_module.html#parameter-remote_src
Or am I missing something?

File: roles/sync/tasks/main.yml (hunk 2 above)
Comment (alxgomz) [reply roles/sync/tasks/main.yml:32]: I'm not sure I get your point.
To me this is needed: https://docs.ansible.com/ansible/latest/collections/ansible/builtin/copy_module.html#parameter-remote_src
Or am I missing something?

File: roles/sync/tasks/main.yml (hunk 1 above)
Comment (gionn) [reply roles/sync/tasks/main.yml:19]: sorry, I was wrong on the meaning of that attribute - I was thinking exactly the opposite (the file has been downloaded right before - it isn't 'remote')

**Pull request merged**